src/libraries/System.Private.CoreLib/src/System/BitConverter.cs

```diff
@@ -742,7 +742,7 @@ public static bool ToBoolean(byte[] value, int startIndex)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);
             if (startIndex < 0)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.startIndex, ExceptionResource.ArgumentOutOfRange_IndexMustBeLess);
-            if (startIndex >= value.Length)
+            if (startIndex > value.Length - 1)
```

Comment:
Wouldn't it be even better as:
```C#
public static bool ToBoolean(byte[] value, int startIndex)
{
    if (value == null)
        ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);
    if ((uint)startIndex >= (uint)value.Length)
        ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.startIndex, ExceptionResource.ArgumentOutOfRange_IndexMustBeLess); // differs from other overloads, which throw base ArgumentException

    return value[startIndex] != 0;
}
```
?

Reply:
I've just measured performance(first one is my fix `i < len - 1` and the second is your `(uint)i <= (uint)len`):


|                                    Method |        Job |                                                                                                  Toolchain |       Mean |      Error |     StdDev |     Median |        Min |        Max | Ratio | RatioSD | Allocated | Alloc Ratio |
|------------------------------------------ |----------- |----------------------------------------------------------------------------------------------------------- |-----------:|-----------:|-----------:|-----------:|-----------:|-----------:|------:|--------:|----------:|------------:|
|                    ReadStructAndReverseBE | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  11.815 ns |  0.7519 ns |  0.8045 ns |  11.585 ns |  11.039 ns |  14.159 ns |  0.90 |    0.08 |         - |          NA |
|                    ReadStructAndReverseBE | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  13.176 ns |  0.5972 ns |  0.6878 ns |  13.636 ns |  11.707 ns |  13.698 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
|                    ReadStructAndReverseLE | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |   1.705 ns |  0.0171 ns |  0.0152 ns |   1.711 ns |   1.679 ns |   1.729 ns |  0.71 |    0.04 |         - |          NA |
|                    ReadStructAndReverseLE | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |   2.430 ns |  0.1144 ns |  0.1272 ns |   2.493 ns |   2.157 ns |   2.504 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
|                  ReadStructFieldByFieldBE | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  10.553 ns |  0.0895 ns |  0.0838 ns |  10.576 ns |  10.335 ns |  10.628 ns |  0.90 |    0.01 |         - |          NA |
|                  ReadStructFieldByFieldBE | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  11.779 ns |  0.0273 ns |  0.0255 ns |  11.777 ns |  11.734 ns |  11.825 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
|                  ReadStructFieldByFieldLE | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |   9.822 ns |  0.0316 ns |  0.0264 ns |   9.825 ns |   9.771 ns |   9.868 ns |  0.99 |    0.00 |         - |          NA |
|                  ReadStructFieldByFieldLE | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |   9.918 ns |  0.0251 ns |  0.0222 ns |   9.921 ns |   9.878 ns |   9.951 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
| ReadStructFieldByFieldUsingBitConverterLE | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  21.574 ns |  0.0554 ns |  0.0491 ns |  21.571 ns |  21.477 ns |  21.655 ns |  0.99 |    0.00 |         - |          NA |
| ReadStructFieldByFieldUsingBitConverterLE | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  21.837 ns |  0.1026 ns |  0.0910 ns |  21.821 ns |  21.693 ns |  22.042 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
| ReadStructFieldByFieldUsingBitConverterBE | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  30.139 ns |  0.2686 ns |  0.2097 ns |  30.031 ns |  29.902 ns |  30.538 ns |  0.97 |    0.01 |         - |          NA |
| ReadStructFieldByFieldUsingBitConverterBE | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun |  31.140 ns |  0.0370 ns |  0.0309 ns |  31.142 ns |  31.084 ns |  31.185 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
|                  MeasureReverseEndianness | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun | 458.299 ns |  2.5360 ns |  2.1177 ns | 457.619 ns | 456.723 ns | 463.796 ns |  1.04 |    0.00 |         - |          NA |
|                  MeasureReverseEndianness | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun | 439.942 ns |  2.4409 ns |  2.2832 ns | 438.691 ns | 437.401 ns | 444.206 ns |  1.00 |    0.00 |         - |          NA |
|                                           |            |                                                                                                            |            |            |            |            |            |            |       |         |           |             |
|                   MeasureReverseUsingNtoH | Job-EGIAJP | /runtime.revert/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun | 452.869 ns | 11.7559 ns | 13.5382 ns | 459.865 ns | 425.363 ns | 461.152 ns |  0.95 |    0.03 |         - |          NA |
|                   MeasureReverseUsingNtoH | Job-MGYUUP |        /runtime/artifacts/bin/testhost/net7.0-Linux-Release-x64/shared/Microsoft.NETCore.App/7.0.0/corerun | 475.243 ns |  4.1602 ns |  3.8915 ns | 475.732 ns | 470.265 ns | 482.813 ns |  1.00 |    0.00 |         - |          NA |

Reply:
Where is BitConverter.ToBoolean used in these tests?

Reply:
Yeah, sorry. These tests:
![image](https://user-images.githubusercontent.com/32279961/165779876-e33942d9-6066-469a-acd9-e554c7469ee0.png)